core/src/main/java/org/apache/calcite/sql2rel/SqlToRelConverter.java

```diff
@@ -4406,9 +4406,29 @@ private RelNode convertDelete(SqlDelete call) {
 
   private RelNode convertCreateTable(SqlCreateTable call) {
 
-    // NOTE: We currently require a SqlCreateTable to have a query in valiation,
+    // NOTE: We currently require a SqlCreateTable to have a query in validation,
     // so the call to requireNonNull is valid here
-    RelRoot relRoot = convertQueryRecursive(requireNonNull(call.getQuery()), false, null);
+    //
+    // NOTE2: Calcite convertQueryRecursive will omit sorts in the logicalPlan when they
```

Comment:
I don't understand this

Reply:
I expanded the comment, hopefully it's clearer now.

Reply:
@jsternberg I want you to specifically check this comment to confirm if it makes sense to you.

Reply:
I'm not completely sure I understand it myself as I don't understand the details. If we have tests asserting that this works as expected and we have a long comment indicating the original intention, I think that's fine because, if it's wrong, someone will be able to identify that it's wrong.

We'll likely be revisiting this area anyway. Part of the reason why this ends up being a bit confusing to me is I'm not sure what is original Calcite code and what is stuff that we added just from this PR by itself. When we separate our changes to the validator into its own class, that will all be a bit more obvious.

Something to also consider is that this might be too much logic in the sql to rel converter. If the goal here is to eliminate unneeded sorts for the purposes of optimization, that seems like something the planner should be able to figure out. You might want to consider just allowing the extra unneeded sort nodes and then removing them later. I guess are the sorts an optimization issue or are they a correctness issue?

Reply:
To clarify, omitting the sorts is an optimization that already exists in Calcite, but we run into a bug when using it. The current implementation is just to allow the extra unneeded sort nodes.

I've updated the comment slightly to make it clearer.